flit_core/tests_core/test_config.py

```diff
@@ -178,3 +198,28 @@ def test_bad_pep621_readme(readme, err_match):
     }
     with pytest.raises(config.ConfigError, match=err_match):
         config.read_pep621_metadata(proj, samples_dir / 'pep621')
+
+
+@pytest.mark.parametrize(('value', 'files'), [
+    ('[]', []),
+    ('["LICENSE"]', ["LICENSE"]),
+    ('["LICENSE*"]', ["LICENSE"]),
+    ('["**/LICENSE*"]', ["LICENSE", "module/vendor/LICENSE_VENDOR"]),
+    ('["module/vendor/LICENSE*"]', ["module/vendor/LICENSE_VENDOR"]),
+    ('["LICENSE", "module/**/LICENSE*"]', ["LICENSE", "module/vendor/LICENSE_VENDOR"]),
+])
+def test_pep621_license_files(value, files):
+    path = samples_dir / 'pep621_license_files' / 'pyproject.toml'
+    data = path.read_text()
+    data = re.sub(
+        r"(^license-files = )(?:\[.*\])", r"\g<1>{}".format(value),
+        data, count=1, flags=re.M
+    )
+    dir = os.getcwd()
+    try:
+        os.chdir(samples_dir / 'pep621_license_files')
+        with patch("pathlib.Path.read_text", return_value=data):
```

Comment:
Can I ask that we test with `read_pep621_metadata()` or `prep_toml_config()`, which both take dicts as input, rather than mocking the file read and regexing replacement data into place? I made a similar change in #712 (6d60848cea0d64a4ec69a55441bf717da9e7fcfc) if it's not clear what I mean. We don't need to test the TOML parser here.